### ISSUE TITLE
Project.ini file from command line results doesn't load DB

### ISSUE THREAD

**Issue by meder411** [colmap/colmap#11]
When running a reconstruction using the [command-line instructions given in the documentation](https://colmap.github.io/cli.html), everything works fine. However, when I want to visualize the resulting sparse reconstruction, or even just manage the DB using the GUI, I can't just load the \*.ini file that is created during the command-line sequence. Going to **File >> Open Project >> Select project.ini** does appear to open the project. However, when one clicks **Processing >> Manage Database**, an error appears saying _"You must create or open a valid project."_ Additionally trying to restart the reconstruction gives the same error.

A work-around is to go to **File >> New Project**, click **Open** for the _Database_ field, and select the \*.db file for the desired project. Then click **Select** for the _Images_ field and select the directory containing the images for that project. Then click **Create**. This at least properly loads the DB, but to visualize the reconstruction, you must run **Reconstruction >> Start / resume reconstruction** all over again.

At least the features and inliers are loaded with the DB, saving you the lion's share of the reconstruction time, but it's frustrating having to register the cameras again in the GUI a second time. That being said, the registration isn't stored in the DB so it may be more of a feature request than a bug report.

Basically what this bug boils down to is that the _project.ini_ file that is created when running the pipeline from the command line does not properly reference the database. Yes, the user can make it work, but it should just be loaded with the project file.

**Comment by ahojnnes** [comment]
Hi Marc,

As described in the documentation (https://colmap.github.io/tutorial.html#exporting-results), you have to import the model to visualize it. Nevertheless, opening the project should work for managing the project. Please, share your sequence of commands with exact paths and the resulting `project.ini` file generated by the mapper.

Thanks for the report,
Johannes

**Comment by meder411** [comment]
Ah crap, I actually knew that about visualization. My bad on that one.

Here is the series of commands:

```
/playpen/colmap/build/src/exe/database_creator --General.database_path colmap/GOPR0297/GOPR0297.db

/playpen/colmap/build/src/exe/feature_extractor --General.database_path colmap/GOPR0297/GOPR0297.db --General.image_path ./GOPR0297 --ExtractionOptions.camera_model OPENCV --ExtractionOptions.single_camera 1 --ExtractionOptions.camera_params 1194.505955,1194.375495,639.754357,343.475637,-0.274884,0.113541,0.0,0.0

/playpen/colmap/build/src/exe/sequential_matcher --General.database_path colmap/GOPR0297/GOPR0297.db --MatchOptions.gpu_index 0 --no_opengl 1

/playpen/colmap/build/src/exe/mapper --General.database_path colmap/GOPR0297/GOPR0297.db --General.image_path ./GOPR0297 --export_path colmap/GOPR0297
```

The _project.ini_ file I subsequently try to load in the GUI is **colmap/GOPR0297/0/project.ini**

**Comment by ahojnnes** [comment]
This is likely due to the fact that you are using relative paths on the command line and the GUI will look relative to the colmap executable path, which is not equivalent to the directory from which you run the commands.

**Comment by meder411** [comment]
I checked wit absolute paths, and , per usual, you are correct.

To the visualization question: how does one export the .ply file from the command line. There does not appear to be an option for that. Thus it does seem the case that there is no way to visualize a command-line-run reconstruction in Colmap without re-running the registration in the GUI. Is this true?

**Comment by ahojnnes** [comment]
I refer you to my first answer: import the model into the GUI and
you'll be able to visualize the model as usual, that is click File >
Import Model > Select the folder of the reconstruction, e.g.
/path/to/results/0/. Other than that, there is a binary called
model_converter that lets you convert to a different format.

**Comment by ahojnnes** [comment]
@meder411 Closing this issue as resolved. Feel free to reopen if you have remaining questions.
